client/src/layouts/navBar/NavBar.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import queryString from 'query-string';
+import { Redirect, Link } from 'react-router-dom';
+import { withStyles } from '@material-ui/core/styles';
+import { BrokenImage } from '@material-ui/icons';
+import Grid from '@material-ui/core/Grid';
+import AppBar from '@material-ui/core/AppBar';
+
+const styles = ({
+  container: {
+    padding: 20,
+  },
+  link: {
+    color: '#fff',
+    textDecoration: 'none',
+    textTransform: 'uppercase',
+  },
+});
+
+class NavBar extends React.PureComponent {
+  componentWillMount() {
+    const { props } = this;
+    const { location } = props;
+    const token = localStorage.getItem('token') || queryString.parse(location.search).token;
+    if (token) {
+      props.checkToken(token);
+    } else {
+      props.push('/sign-in');
+    }
+  }
+
+  render() {
+    const { props } = this;
+    const { classes } = props;
+    const { responseCheckToken } = props;
```

Comment:
you can use nested destructuring here, for example : const { props : { classes, responseCheckToken } = {} } = this